File: maths/dice_throw.js
/**
 * Put into a plotly bar codepen
 * @param {*} n 
 * @returns 
 */
function toss(n) {
    var result = []
    for(var i = 0; i< n; i++){
      var values = [0, 1, 2]
      var randomElement = Math.floor(Math.random() * values.length)
      result.push(randomElement)
    }
    return result
}

function run(n, d){
	var result = {}
	for (var i = 0; i<=d; i++){
		result[i] = 0
  }
	for (var i = 0; i <n; i++){
		var t = toss(d) 
		var numOfOne = t.filter(x => x==1).length
    result[numOfOne] = result[numOfOne] + 1
  }
  return result
}

var result = run(100000, 10)
var data = [
  {
    x: Object.keys(result),
    y: Object.values(result),
    type: 'bar'
  }
];


Plotly.newPlot('myDiv', data);

File: maths/dice_throw.py
import random
import matplotlib.pyplot as plt
def toss(n):
    result = []
    for i in range(n):
        result.append(random.choice([0, 1, 2]))
    #print(result)
    return result

# d is the number of dice rolled
# n is the total number of the coin toss experiment to run to generate the histogram plot
def run(n, d):
	result = {}
	for i in range(d):
		result[i] = 0
	for i in range(n):
		t = toss(d)
		numOfOne = t.count(1)
		result[numOfOne] = result.get(numOfOne, 0) + 1

	print(result)
	plt.bar(result.keys(), result.values())
	plt.show()

run(100000, 10)